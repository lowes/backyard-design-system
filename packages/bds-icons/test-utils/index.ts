import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
