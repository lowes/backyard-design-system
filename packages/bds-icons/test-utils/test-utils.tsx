import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
