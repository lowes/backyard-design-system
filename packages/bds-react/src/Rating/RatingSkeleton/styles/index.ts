import styled from 'styled-components'

const RatingSkeletonWrapper = styled.span`
    display: inline-flex;
    flex-direction: row;
    align-items: center;

    &.size--jumbo {
        .variant--circle { margin-right: ${({ theme }) => theme.sizes.size_4}; }
        .variant--rect { margin-left: ${({ theme }) => theme.sizes.size_10}; }
    }

    &.size--large {
        .variant--circle { margin-right: ${({ theme }) => theme.sizes.size_2}; }
        .variant--rect { margin-left: ${({ theme }) => theme.sizes.size_6}; }
    }

    &.size--medium {
        .variant--rect { margin-left: ${({ theme }) => theme.sizes.size_2}; }
    }

    &.size--small {
        .variant--rect { margin-left: ${({ theme }) => theme.sizes.size_2}; }
    }
`

export { RatingSkeletonWrapper }
export default RatingSkeletonWrapper