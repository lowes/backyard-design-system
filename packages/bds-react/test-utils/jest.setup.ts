import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
