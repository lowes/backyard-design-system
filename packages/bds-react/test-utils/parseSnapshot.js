const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
