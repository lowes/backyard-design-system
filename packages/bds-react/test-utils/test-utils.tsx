import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
