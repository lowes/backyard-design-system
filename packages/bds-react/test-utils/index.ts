// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
