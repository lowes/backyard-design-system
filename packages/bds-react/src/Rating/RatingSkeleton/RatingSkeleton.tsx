import * as React from 'react'
import classnames from 'classnames'
import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'

import { RatingSkeletonWrapper } from './styles'

import Skeleton from '../../Skeleton'

/**
 * Backyard's Ratings Skeleton component
 * 
 * Returns a set of Skeleton circles which each has a size determined
 * by the value of the size prop. Optional paramter of `withValue` to
 * determine if a Skeleton rect for a value should be shown. The label
 * shows scales in size determined by the value of the size prop also.
 * 
 * <RatingSkeleton />
 */

const RatingSkeleton: BDSForwardRef<RatingSkeletonProps> = React.forwardRef<
    RatingSkeletonRef,
    RatingSkeletonProps
>(function RatingSkeleton (
    {
        size = 'medium',
        withValue = false,
        animate = false,
        classNames,
        ...props
    },
    ref,
) {

    let starHeight = 'size_16'
    let starWidth = 'size_16'
    let textHeight = 'size_20'
    let textWidth = 'size_32'

    switch(size) {
        case 'jumbo':
            starHeight = 'size_32'
            starWidth = 'size_32'
            textHeight = 'size_28'
            textWidth = 'size_44'
            break

        case 'large':
            starHeight = 'size_24'
            starWidth = 'size_24'
            textHeight = 'size_24'
            textWidth = 'size_36'
            break

        case 'small':
            textHeight = 'size_18'
            textWidth = 'size_28'
            break

        default:
            starHeight = 'size_24'
            starWidth = 'size_24'
            textHeight = 'size_24'
            textWidth = 'size_36'
            break
    }

    return (
        <RatingSkeletonWrapper className={classnames(classNames, 'rating--skeleton', `size--${size}`)} {...props} ref={ref}>
            <Skeleton variant='circle' height={starHeight} width={starWidth} animate={animate} />
            <Skeleton variant='circle' height={starHeight} width={starWidth} animate={animate} />
            <Skeleton variant='circle' height={starHeight} width={starWidth} animate={animate} />
            <Skeleton variant='circle' height={starHeight} width={starWidth} animate={animate} />
            <Skeleton variant='circle' height={starHeight} width={starWidth} animate={animate} />
            {(withValue) ? 
                <Skeleton variant='rect' height={textHeight} width={textWidth} animate={animate} /> :
                null
            }
        </RatingSkeletonWrapper>
    )
})

type RatingSkeletonRef = HTMLSpanElement

interface RatingSkeletonProps {
    animate?: boolean 
    size?: string
    withValue?: boolean
    classNames?: string
}

RatingSkeleton.bdsName = 'RatingSkeleton'

export { RatingSkeleton }

export type { RatingSkeletonProps, RatingSkeletonRef }

export default RatingSkeleton