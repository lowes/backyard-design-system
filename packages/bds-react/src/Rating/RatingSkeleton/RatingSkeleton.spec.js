import * as React from 'react'
import { render } from '../../../test-utils'
import RatingSkeleton from './RatingSkeleton'

describe('RatingSkeleton Snapshots', () => {
    test('RatingSkeleton default snapshot', () => {
        const { asFragment } = render(
            <RatingSkeleton />
        )

        expect(asFragment()).toMatchSnapshot()
    })


    test('RatingSkeleton animate snapshot', () => {
        const { asFragment } = render(
            <RatingSkeleton animate={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('RatingSkeleton sizes snapshot', () => {
        const sizes = ['small', 'medium', 'large', 'jumbo']

        sizes.forEach(size => {
            const { asFragment } = render(
                <RatingSkeleton size={size} />
            )
    
            expect(asFragment()).toMatchSnapshot()
        })
    })
})