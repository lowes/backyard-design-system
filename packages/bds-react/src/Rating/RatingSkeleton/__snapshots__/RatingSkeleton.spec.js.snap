// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingSkeleton Snapshots RatingSkeleton animate snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.size--jumbo .variant--circle {
  margin-right: 0.25rem;
}

.c0.size--jumbo .variant--rect {
  margin-left: 0.625rem;
}

.c0.size--large .variant--circle {
  margin-right: 0.125rem;
}

.c0.size--large .variant--rect {
  margin-left: 0.375rem;
}

.c0.size--medium .variant--rect {
  margin-left: 0.125rem;
}

.c0.size--small .variant--rect {
  margin-left: 0.125rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.5rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.5rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<span
    class="c0 rating--skeleton size--medium"
  >
    <div
      class="c1 variant--circle shape--rounded style--animated"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--animated"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--animated"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--animated"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--animated"
      height="size_24"
      width="size_24"
    />
  </span>
</DocumentFragment>
`;

exports[`RatingSkeleton Snapshots RatingSkeleton default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.size--jumbo .variant--circle {
  margin-right: 0.25rem;
}

.c0.size--jumbo .variant--rect {
  margin-left: 0.625rem;
}

.c0.size--large .variant--circle {
  margin-right: 0.125rem;
}

.c0.size--large .variant--rect {
  margin-left: 0.375rem;
}

.c0.size--medium .variant--rect {
  margin-left: 0.125rem;
}

.c0.size--small .variant--rect {
  margin-left: 0.125rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.5rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.5rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<span
    class="c0 rating--skeleton size--medium"
  >
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
  </span>
</DocumentFragment>
`;

exports[`RatingSkeleton Snapshots RatingSkeleton sizes snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.size--jumbo .variant--circle {
  margin-right: 0.25rem;
}

.c0.size--jumbo .variant--rect {
  margin-left: 0.625rem;
}

.c0.size--large .variant--circle {
  margin-right: 0.125rem;
}

.c0.size--large .variant--rect {
  margin-left: 0.375rem;
}

.c0.size--medium .variant--rect {
  margin-left: 0.125rem;
}

.c0.size--small .variant--rect {
  margin-left: 0.125rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1rem;
  height: 1rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1rem;
  height: 1rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<span
    class="c0 rating--skeleton size--small"
  >
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_16"
      width="size_16"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_16"
      width="size_16"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_16"
      width="size_16"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_16"
      width="size_16"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_16"
      width="size_16"
    />
  </span>
</DocumentFragment>
`;

exports[`RatingSkeleton Snapshots RatingSkeleton sizes snapshot 2`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.size--jumbo .variant--circle {
  margin-right: 0.25rem;
}

.c0.size--jumbo .variant--rect {
  margin-left: 0.625rem;
}

.c0.size--large .variant--circle {
  margin-right: 0.125rem;
}

.c0.size--large .variant--rect {
  margin-left: 0.375rem;
}

.c0.size--medium .variant--rect {
  margin-left: 0.125rem;
}

.c0.size--small .variant--rect {
  margin-left: 0.125rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.5rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.5rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<span
    class="c0 rating--skeleton size--medium"
  >
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
  </span>
</DocumentFragment>
`;

exports[`RatingSkeleton Snapshots RatingSkeleton sizes snapshot 3`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.size--jumbo .variant--circle {
  margin-right: 0.25rem;
}

.c0.size--jumbo .variant--rect {
  margin-left: 0.625rem;
}

.c0.size--large .variant--circle {
  margin-right: 0.125rem;
}

.c0.size--large .variant--rect {
  margin-left: 0.375rem;
}

.c0.size--medium .variant--rect {
  margin-left: 0.125rem;
}

.c0.size--small .variant--rect {
  margin-left: 0.125rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.5rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.5rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<span
    class="c0 rating--skeleton size--large"
  >
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_24"
      width="size_24"
    />
  </span>
</DocumentFragment>
`;

exports[`RatingSkeleton Snapshots RatingSkeleton sizes snapshot 4`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.size--jumbo .variant--circle {
  margin-right: 0.25rem;
}

.c0.size--jumbo .variant--rect {
  margin-left: 0.625rem;
}

.c0.size--large .variant--circle {
  margin-right: 0.125rem;
}

.c0.size--large .variant--rect {
  margin-left: 0.375rem;
}

.c0.size--medium .variant--rect {
  margin-left: 0.125rem;
}

.c0.size--small .variant--rect {
  margin-left: 0.125rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 2rem;
  height: 2rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 2rem;
  height: 2rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<span
    class="c0 rating--skeleton size--jumbo"
  >
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_32"
      width="size_32"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_32"
      width="size_32"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_32"
      width="size_32"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_32"
      width="size_32"
    />
    <div
      class="c1 variant--circle shape--rounded style--default"
      height="size_32"
      width="size_32"
    />
  </span>
</DocumentFragment>
`;
